modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -1021,7 +1024,7 @@ private void stopCache(GridCacheAdapter<?, ?> cache, boolean cancel, boolean des
 
             sharedCtx.removeCacheContext(ctx);
 
-            cache.stop();
+            cache.stop(destroy);
```

Comment:
It seems this place (and below at line 1041) are only places where we should remove metric configuration. IF this code will be invoked from this place then we don't need any additional flags for `GridCacheAdapter.stop()` method.

Reply:
Fixed.